pkg/backup/volume_snapshot_action.go

```diff
@@ -60,6 +61,13 @@ func (a *volumeSnapshotAction) Execute(ctx ActionContext, volume map[string]inte
 
 	metadata := volume["metadata"].(map[string]interface{})
 	name := metadata["name"].(string)
+	var pvfailureDomainZone string
+	labelsMap, err := collections.GetMap(metadata, "labels")
+	if err == nil {
+		pvfailureDomainZone = labelsMap["failure-domain.beta.kubernetes.io/zone"].(string)
```

Comment:
need to do a nil-check here when getting `failure-domain.beta.kubernetes.io/zone` out of the map to handle the case where this label doesn't exist (currently it panics when it tries to convert to string). Can you also make sure each situation has an appropriate log statement (no labels collection, no zone key, or zone found)?